cli/pack/common.go

```diff
@@ -108,33 +111,59 @@ func detectVersion(ctx *context.Ctx) error {
 	return nil
 }
 
-func getPackageFullname(ctx *context.Ctx) string {
-	ext, found := extByType[ctx.Pack.Type]
-	if !found {
-		panic(project.InternalError("Unknown type: %s", ctx.Pack.Type))
+func getPackageSuffix(ctx *context.Ctx) string {
+	var suffix string
+	if ctx.Pack.Suffix != "" {
+		suffix = fmt.Sprintf("-%s", ctx.Pack.Suffix)
 	}
 
-	packageFullname := fmt.Sprintf(
-		"%s-%s",
+	return suffix
+}
+
+func getRpmPackageNameBody(ctx *context.Ctx) string {
+	return fmt.Sprintf(
+		"%s-%s%s.%s",
 		ctx.Project.Name,
 		ctx.Pack.VersionRelease,
+		getPackageSuffix(ctx),
+		runtime.GOARCH,
 	)
+}
 
-	if ctx.Pack.Suffix != "" {
-		packageFullname = fmt.Sprintf(
-			"%s-%s",
-			packageFullname,
-			ctx.Pack.Suffix,
-		)
+func getDebPackageNameBody(ctx *context.Ctx) string {
+	return fmt.Sprintf(
+		"%s_%s-%s%s_%s",
+		ctx.Project.Name,
+		ctx.Pack.VersionRelease,
+		debianVersion,
+		getPackageSuffix(ctx),
+		runtime.GOARCH,
```

Comment:
When building in docker, this value will depend on the image architecture, so using `runtime` isn't honest here.

Reply:
Well, why it isn't honest? Docker container uses the kernel and CPU of the host machine (for example, see [this question](https://www.edureka.co/community/65179/do-docker-containers-have-their-own-kernel-and-cpu)). If docker runs under VM, `runtime.GOARCH` will be the same as in VM, but package will also be built under VM setup. It seems like it is fair in all of the cases.

Reply:
Use `noarch`/`all` to not imply any new restrictions on package, see #675 for extension of this idea

Reply:
I guess Liza pointing out that docker can run an image from a foreign architecture using qemu. Anyway, it is irrelevant now and I'll resolve this discussion.